bin/node/src/node/actors/derivation.rs

```diff
@@ -2,35 +2,231 @@
+                                        }
+                                        .signal(),
+                                    )
+                                    .await?;
+                            } else {
+                                if let ResetError::ReorgDetected(expected, new) = e {
+                                    warn!(
+                                        target: "derivation",
+                                        "L1 reorg detected! Expected: {expected} | New: {new}"
+                                    );
+                                }
+
+                                // Reset the pipeline to the initial L2 safe head and L1 origin,
+                                // and try again.
+                                let l1_origin = self
+                                    .pipeline
+                                    .origin()
+                                    .ok_or(PipelineError::MissingOrigin.crit())?;
+                                self.pipeline
+                                    .signal(
+                                        ResetSignal {
+                                            l2_safe_head: self.l2_safe_head,
+                                            l1_origin,
+                                            system_config: Some(system_config),
+                                        }
+                                        .signal(),
+                                    )
+                                    .await?;
+                            }
+                        }
+                        PipelineErrorKind::Critical(_) => {
+                            error!(target: "derivation", "Critical derivation error: {e}");
+                            return Err(e.into());
+                        }
+                    }
+                }
+            }
+
+            // If there are any new attributes, send them to the execution actor.
+            if let Some(attrs) = self.pipeline.next() {
+                return Ok(attrs);
+            }
+        }
+    }
 }
 
 #[async_trait]
 impl<P> NodeActor for DerivationActor<P>
 where
-    P: Pipeline + OriginAdvancer + SignalReceiver + Send + Sync,
+    P: Pipeline + SignalReceiver + Send + Sync,
 {
-    type Error = ();
+    type Error = DerivationError;
     type Event = NodeEvent;
 
-    async fn start(self) -> Result<(), Self::Error> {
-        unimplemented!()
+    async fn start(mut self) -> Result<(), Self::Error> {
+        loop {
+            select! {
+                _ = self.cancellation.cancelled() => {
+                    info!(
+                        target: "derivation",
+                        "Received shutdown signal. Exiting derivation task."
+                    );
+                    return Ok(());
+                }
+                msg = self.receiver.recv() => {
+                    self.process(msg?).await?;
+                }
```

Comment:
This only makes sense to block here since the channel can fill up and tell the caller if it is ready to receive new messages?